book/src/12_tracked.md

```diff
@@ -30,73 +30,52 @@ impl Component for Data {
 #[derive(Default)]
 pub struct Sys {
     pub dirty: BitSet,
-    pub modify_reader_id: Option<ReaderId<ModifiedFlag>>,
-    pub insert_reader_id: Option<ReaderId<InsertedFlag>>,
+    pub reader_id: Option<ReaderId<ComponentEvent>>,
 }
 
 impl<'a> System<'a> for Sys {
     type SystemData = (
         ReadStorage<'a, Data>,
         WriteStorage<'a, SomeOtherData>,
     );
-    
-    fn run(&mut self, (data, some_other_data): Self::SystemData) {
+
+    fn run(&mut self, (data, mut some_other_data): Self::SystemData) {
         self.dirty.clear();
-        
-        data.populate_inserted(&mut self.insert_reader_id.as_mut().unwrap(), &mut self.dirty);
-        
-        // Note that we could use separate bitsets here, we only use one to simplify the example
-        data.populate_modified(&mut self.modify_reader_id.as_mut().unwrap(), &mut self.dirty);
-        
-        for (d, other, _) in (&data, &mut some_other_data, &self.dirty) {
+
+        let events = data.channel().read(self.reader_id.as_mut().unwrap());
+
+        // Note that we could use separate bitsets here, we only use one to
+        // simplify the example
+        for event in events {
+            match event {
+                ComponentEvent::Modified(id) | ComponentEvent::Inserted(id) => {
+                    self.dirty.add(*id);
+                }
+                ComponentEvent::Removed(_) => (),
```

Comment:
I think another comment here would be good:

```suggestion
                 // We don't need to take this event into account since
                 // removed components will be filtered out by the join;
                 // if you want to, you can use `self.dirty.remove(*id);`
                 // so the bit set only contains IDs that still exist
                 ComponentEvent::Removed(_) => (),
```